src/Tooltip/Tooltip.tsx

```diff
@@ -53,68 +112,162 @@ export class Tooltip extends React.Component<TooltipProps, TooltipState> {
-                    onClick={(e: React.MouseEvent<HTMLDivElement>) => !this.props.triggerOnHover && this.toggleTooltip(undefined, e)}
-                    onMouseEnter={() => this.props.triggerOnHover && this.toggleTooltip(true)}
-                    onMouseLeave={() => this.props.triggerOnHover && this.toggleTooltip(false)}
+                    className="tooltip-reference"
+                    aria-describedby={this.state.id}
+                    tabIndex={-1}
+                    onClick={(e: React.MouseEvent<HTMLDivElement>) => ((!this.props.trigger && !this.props.triggerOnHover) || this.props.trigger === "click") && this.onTooltipToggle(e)}
+                    onMouseEnter={(e: React.MouseEvent<HTMLDivElement>) => (this.props.triggerOnHover || this.props.trigger === "hover") && this.onTooltipToggle(e, true)}
+                    onMouseLeave={(e: React.MouseEvent<HTMLDivElement>) => (this.props.triggerOnHover || this.props.trigger === "hover") && this.onTooltipToggle(e, false)}
+                    onFocus={(e: React.FocusEvent<HTMLDivElement>) => this.props.trigger === "focus" && this.onTooltipToggle(e)}
+                    onBlur={(e: React.FocusEvent<HTMLDivElement>) => this.onTooltipToggle(e, false)}
                 >
-                    {this.props.customSvg ? this.props.customSvg : InfoCircleIcon}
-                </div>
-                <div className={`content ${this.props.position || "bottom"} ${this.props.theme || "default"}${this.state.toggle ? " open" : ""}`}>
-                    {this.isPositioned("bottom") && <div className="triangle" />}
-
-                    {!this.props.messageGroup &&
-                        <div className="message-container" style={{ width: `${this.props.width || 120}px` }}>
-                            {this.props.title && <div className="title">{this.props.title}</div>}
-                            <div className="message">{this.props.message || "Tooltip is empty. Please pass a message."}</div>
-                        </div>
-                    }
-
-                    {this.props.messageGroup &&
-                        <div className="message-container" style={{ width: `${this.props.width || 120}px` }}>
-                            {this.props.messageGroup.map((item, index) =>
-                                <div key={index} className="message-list-item">
-                                    {item.title && <div className="title">{item.title}</div>}
-                                    {item.message && <div className="message">{item.message}</div>}
-                                </div>
-                            )}
-                        </div>
-                    }
-
-                    {(this.isPositioned("top") || this.isPositioned("right") || this.isPositioned("left")) && <div className="triangle" />}
+                    {this.props.children || <div className="icon">{this.props.customSvg ? this.props.customSvg : InfoCircleIcon}</div>}
```

Comment:
We don't want to assume it's an icon. We want to start using the tooltip with any UI element. It could be an input element or anything else

Reply:
Ya, we do not assume it as an icon. Users are now encouraged to define the tooltip reference as the children of the component

Reply:
I was talking about the naming, don’t call it icon. The classname and the prop name are both called icon

Reply:
rename classname to default-content